### PR TITLE
Issue 13720 not edit system fields

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts [hunk 1]
```diff
@@ -119,15 +134,25 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
         this.setAutoCheckValues();
     }
 
+    private isPropertyDisabled(property: string): boolean {
+        return (
+            this.fieldPropertyService.isDisabledInEditMode(property) ||
+            this.fieldPropertyService.isDisabledInFixed(this.formFieldData.fixed, property)
```
Comment (fmontes) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: In order to keep the pattern we were handling here, I think is better to do

```|| this.formFieldData.fixed && this.fieldPropertyService.isDisabledInFixed(property)```

But I want comments from @alfredo-dotcms and @freddyucv on this.

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/name-property/name-property.component.html [hunk 2]
```diff
@@ -1,11 +1,11 @@
 <div [formGroup]="group">
```
Comment (fmontes) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/name-property/name-property.component.html]: Don't pass prettier in the html

Comment (hmoreras) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/field-properties/name-property/name-property.component.html]: I didn't :), some white spaces where removed

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts [hunk 3]
```diff
@@ -21,7 +31,11 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
     fieldProperties: string[] = [];
     checkboxFields: string[] = ['indexed', 'listed', 'required', 'searchable', 'unique'];
 
-    constructor(private fb: FormBuilder, public dotMessageService: DotMessageService, private fieldPropertyService: FieldPropertyService) {
+    constructor(
+        private fb: FormBuilder,
```
Comment (jdotcms) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: rename to formBuilder

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts (hunk 1 above)
Comment (jdotcms) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: it looks more such as business question!

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts [hunk 4]
```diff
@@ -119,15 +134,25 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
...
+        );
+    }
+
     private sortProperties(properties: string[]): void {
         this.fieldProperties = properties
-            .filter((property) => this.fieldPropertyService.existsComponent(property))
-            .sort((property1, proeprty2) => this.fieldPropertyService.getOrder(property1) - this.fieldPropertyService.getOrder(proeprty2));
+            .filter(property => this.fieldPropertyService.existsComponent(property))
+            .sort(
+                (property1, proeprty2) =>
```
Comment (jdotcms) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: a typo here!

File: src/app/portlets/content-types/fields/service/field-properties.service.ts [hunk 5]
```diff
@@ -82,6 +82,16 @@ export class FieldPropertyService {
...
     }
 
+    /**
+     * Return true if field is fixed and the property should be disabled when fixed is true
+     * @param {boolean} fixed
+     * @param {string} propertyName
+     * @returns {boolean}
+     */
+    isDisabledInFixed(fixed: boolean, propertyName: string): boolean {
+        return fixed ? (PROPERTY_INFO[propertyName] ? PROPERTY_INFO[propertyName].disabledInFixed : false) : false;
```
Comment (jdotcms) [inline src/app/portlets/content-types/fields/service/field-properties.service.ts]: nested ternary operator is a smell code!

File: src/app/portlets/content-types/fields/service/field-properties.service.spec.ts [hunk 6]
```diff
@@ -104,6 +104,11 @@ describe('FieldPropertyService', () => {
         expect(this.fieldPropertiesService.isDisabledInEditMode('property')).toBeNull();
     });
 
+    it('should return if the property is editable when field is fixed', () => {
+        expect(true).toEqual(this.fieldPropertiesService.isDisabledInFixed( 'name'));
```
Comment (fmontes) [inline src/app/portlets/content-types/fields/service/field-properties.service.spec.ts]: Spaces issues here... ```( '...```

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts [hunk 7]
```diff
@@ -141,8 +166,10 @@ export class ContentTypeFieldsPropertiesFormComponent extends BaseComponent impl
             this.handleDisabledIndexed(true);
         }
 
-        checkbox.valueChanges.subscribe((res) => {
-            checkbox === this.form.get('unique') ? this.handleUniqueValuesChecked(res) : this.setIndexedValueChecked(res);
+        checkbox.valueChanges.subscribe(res => {
```
Comment (fmontes) [inline src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: Please check what's happening with with your prettier looks like is not taking the config.

File: src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts (hunk 4 above)
Comment (fmontes) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: Please fix the typo.

Comment (fmontes) [reply src/app/portlets/content-types/fields/content-type-fields-properties-form/content-type-fields-properties-form.component.ts]: Ans also the Prettier.

**Pull request merged**